### PR TITLE
Setup MailHog in Docker to send test notifications

### LINKED ISSUE #126: Use MailHog in Docker for testing notification emails
Let's add [MailHog](https://github.com/mailhog/MailHog) to our `docker-compose.yml` so we can test sending emails in development and testing.  It has a web UI and also an API for tests to check that things worked.

We'd have to use an alternate transport config in dev vs. production.

See:

- https://technotrampoline.com/articles/how-to-intercept-nodemailer-emails-with-mailhog/
- https://stackoverflow.com/questions/72459370/nodemailer-with-mailhog-using-docker
- https://jfelix.info/blog/test-your-email-flow-with-mailhog-a-fake-smtp-server
- etc

### PR DESCRIPTION
**Pull request by SerpentBytes**
## Description
This PR resolves #126 by setting up MailHog in Docker to test sending notification emails in the development environment.

#### Steps to Test
1. `cp .env.example .env`
2. `npm run docker`
3.  `npm run setup`
4. `npm run build`
5. `npm run dev`
6. Navigate to MailHog UI at `localhost:8025`:
Screenshot 2023-02-07 at 9 43 20 PM

#### Additional Information
There are currently no tests implemented to check whether we are able to send notifications successfully. I intend on implementing tests for notifications in a follow-up PR.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -11,3 +11,7 @@ PORT=8080
 
 # One of: trace, debug, info, warn, error, fatal, silent
 LOG_LEVEL=debug
+
+# Nodemailer config
+NODEMAILER_USER="no-reply@senecacollege.ca"
```
Comment (humphd) [inline .env.example:16]: Let's rename this to `NOTIFICATIONS_EMAIL_USER`.  The nodemailer bit is an internal detail that admins of the system won't care about. Add a comment above it that indicates what it's for.

**Review by humphd — APPROVED** [review]
One suggestion, looks great otherwise.

cc @sirinoks, we should add some docs about how to use this new test email server in development.  It's really cool, you can go to http://localhost:8025 and you have a GMail like UI that shows you the emails you've sent.  There is also an API you can use in testing, which @TD will probably document when he writes tests.

**Review by Genne23v — DISMISSED** [review]
It looks good to me!

File: .env.example (hunk 1 above)
Comment (SerpentBytes) [reply .env.example]: @humphd resolved in latest commit.

**Review by humphd — APPROVED** [review]

**Pull request merged**